services/terrain/src/terrain/persistence/search.clj

```diff
@@ -110,46 +110,55 @@
       (throw+ es-uninitialized))))
 
 
-(defn- tags-access-filter
-  [tags memberships]
-  (letfn [(tag-filter [tag] (query/term :id {:type  "tag"
-                                             :id    tag
-                                             :path  "targets.id"
-                                             :cache false}))]
-    (query/bool :must   (query/nested :path   "userPermissions"
-                                      :filter (query/term "userPermissions.user" memberships))
-                :should (map tag-filter tags))))
+(defn- mk-filter
+  [in-folders tags memberships]
+  (let [filter-path (fn [path] (query/prefix "path" (if (= \/ (last path))
+                                                      path
+                                                      (str path \/))))
+        filter-tag  (fn [tag] (query/term :id {:type  "tag"
+                                               :id    tag
+                                               :path  "targets.id"
+                                               :cache false}))
+        perm-filter (query/nested :path   "userPermissions"
+                                  :filter (query/term "userPermissions.user" memberships))]
+    (query/bool :must   (query/bool :must perm-filter :should (map filter-path in-folders))
+                :should (map filter-tag tags))))
```

Comment:
Cool, thanks. Couldn't find the default for that parameter.

I realize that I also hadn't properly read the nesting, which was part of my confusion, so in case anyone else runs into it, it's `(:must (:must <permissions> :should <in folder>) :should <tags>)` which with the minimum of 1 works out to "must match perms and at least one folder and at least one tag".